tern/utils/general.py

```diff
@@ -182,3 +183,175 @@ def parse_image_string(image_string):
                 'digest_type': tokens[1],
                 'digest': tokens[2]}
     return None
+
+
+def split_shell_script(script, skip_branch=False):
+    """Given a shell script, split it into statement of
+    1. variable
+    2. command
+    3. branch: if, case
+    Use dictionary to store infomations on a statement, it has following keys:
+    - variable(only for variable statement):
+        - name: variable name
+        - value: variable value
+    - variable(only for variable statement):
+        - name: command name(eg. apt-get)
+        - options: command options(eg. ["--no-install-recommends", "-y"])
+        - words: command words
+    - branch(only for variable statement):
+        - type: branch type(if or case)
+        - statements: branch variable and command statements list
+    - content: statement sentence(eg. 'apt-get update')
+
+    sentence is the text we are parsing.(string)
+    statement is the parsed result for sentence.(dictionary)
+    return a list of statements
+    """
+    # add ';' in case that the script is not ended with ';'
+    script += ';'
+    if skip_branch:
```

Comment:
I use skip_branch here since i need to reuse split_shell_script() to parse the sentences in a branch. When we are dealing with sentences in a branch, they are already been splited by shlex, so we do not need to split again.

Reply:
I think you can avoid this by creating another function that deals with the output of `shlex.split(script, posix=False)`.